File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/components/types/AbstractComponent.java
package cz.cvut.fel.matyapav.afandroid.components.types;

import com.tomscz.afswinx.rest.rebuild.holder.AFDataHolder;

import cz.cvut.fel.matyapav.afandroid.enums.SupportedComponents;

/**
 * @author Pavel Matyáš (matyapav@fel.cvut.cz).
 *
 *@since 1.0.0..
 */
public interface AbstractComponent {

    void insertData(String dataResponse, StringBuilder road);

    SupportedComponents getComponentType();

    boolean validateData();

    AFDataHolder reserialize();

}

File: AFMobile/AFWinPhone/AFWinPhone/components/types/AbstractComponent.cs
using AFWinPhone.enums;
using AFWinPhone.rest.holder;
using System;
using System.Text;

namespace AFWinPhone.components.types
{
    interface AbstractComponent
    {
        void insertData(String dataResponse, StringBuilder road);

        SupportedComponents getComponentType();

        bool validateData();

        AFDataHolder reserialize();

    }
}
